src/main/java/seedu/address/logic/parser/AddPersonalTaskCommandParser.java

```diff
@@ -71,24 +72,22 @@ public AddPersonalTaskCommand parse(String task) throws ParseException {
      * @throws DurationParseException if duration format is invalid
      * @throws TimingClashException   if there is a timing clash
      */
-    public static void parseTask(String task) throws DateTimeParseException,
-            DurationParseException, TimingClashException, IllegalValueException {
+    private static void parseTask(String task) throws DateTimeParseException,
+            DurationParseException, TimingClashException {
 
         String[] arguments = task.split("\\s", MAXIMUM_AMOUNT_OF_TASK_PARAMETER);
-        if (!canPassInitialCheck(arguments)) {
-            throw new IllegalValueException(MESSAGE_INVALID_INPUT_FORMAT);
-        }
         parseDescription(arguments);
         try {
             parseDateTime(arguments);
             parseDuration(arguments);
-            Schedule.checkClashes(taskDateTime, duration);
+            isTaskClash(taskDateTime, duration);
         } catch (DateTimeParseException dtpe) {
             throw new DateTimeParseException(MESSAGE_INVALID_DATE_TIME, dtpe.getParsedString(), dtpe.getErrorIndex());
         } catch (DurationParseException dpe) {
```

Comment:
can delete this catch statement since it is also handled in the constructor?

Reply:
the one in constructor is checking for the general format, not the exact input. I dont want to combine them because the message thrown are different